### DISCUSSION TITLE
Temporary tablespace on a node-local ephemeral volume

### DISCUSSION THREAD

**Discussion by wvengen** [cloudnative-pg/cloudnative-pg#5458] [Ideas]
We are running our databases with persistent volumes on network attached storage (NAS). For complex analytical queries that need temporary disk space to run, as well as temporary tables, it may be more appropriate to use node-local ephemeral storage. I did not see documented how to set this up, and wonder why this would not be suggested by default?

I do see an emptydir volume in a database's pod definition, but it doesn't look like it's being used for a temporary tablespace (`SHOW temp_tablespaces` is empty). While not documented in [cloudnative-pg's page on tablespaces](https://cloudnative-pg.io/documentation/current/tablespaces/), it seems to be possible to provide a custom PVC template (as seen [in the CRD](https://github.com/cloudnative-pg/cloudnative-pg/blob/bd86755e192a5682ee73d0a882eafcb8dee9e5f6/config/crd/bases/postgresql.cnpg.io_clusters.yaml#L5010-L5261) and [temporary tablespaces PR](https://github.com/cloudnative-pg/cloudnative-pg/pull/3464)). But I don't think that could be used for an [`emptyDir` volume](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir).

An option that I hoped would work, was to manually create the tablespace in PostgreSQL, and use the existing ephemeral `emptyDir` volume (provided the directory can somehow be auto-created, preferably by cloudnative-pg). But that appears to be a fixed configuration parameter. I _could_ try to go the `hostPath` route, but I'd like to avoid node-specifics, and is a little complex with dynamic workloads (each database node needing to get its own path).

Curious to hear any ideas about this.

**Comment by gbartolini** [comment]
We cannot use volatile volumes for PostgreSQL tablespaces. See: https://www.2ndquadrant.com/en/blog/postgresql-no-tablespaces-on-ramdisks/

**Reply by wvengen** [reply]
Thank you for pointing to the relevant blogpost! A bummer that this is indeed not as straightforward as I'd thought. Hope the temporary tablespace support (similar to as described in the blogpost) will happen at some point (I will see if I can play a role, but it is a bit outside the area I have experience with).
